Sources/SwiftProtobufPluginLibrary/BytecodeWriter.swift

```diff
@@ -0,0 +1,105 @@
+// Sources/SwiftProtobufPluginLibrary/BytecodeCompiler.swift - Internal bytecode compiler
+//
+// Copyright (c) 2014 - 2025 Apple Inc. and the project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See LICENSE.txt for license information:
+// https://github.com/apple/swift-protobuf/blob/main/LICENSE.txt
+//
+// -----------------------------------------------------------------------------
+
+import Foundation
+
+/// Writes bytecode into a string that is suitable for use as a Swift string literal (i.e., for
+/// appending to generated code).
+///
+/// See `SwiftProtobuf.BytecodeInterpreter` for more information on the format used here.
+public struct BytecodeWriter<Instruction: RawRepresentable> where Instruction.RawValue == UInt64 {
+    /// The Swift string literal that represents the written bytecode, including the delimiting
+    /// quotes.
+    public var stringLiteral: String { #""\#(code)""# }
+
+    /// The contents of the Swift string literal representing the written bytecode, without the
+    /// delimiting quotes.
+    private var code: String = ""
+
+    /// Creates a bytecode writer that uses the given program format.
+    ///
+    /// - Parameter programFormat: The program format of the bytecode stream to write. Currently,
+    ///   the only valid value is zero.
+    public init(programFormat: Int = 0) {
```

Comment:
Should there be a constant over in the main runtime library for the zero value that is used by these places also?